pkg/workers/sharings/share_data_test.go

```diff
@@ -141,34 +179,312 @@ func TestDeleteDoc(t *testing.T) {
+	}
+	ts = setup.GetTestServerMultipleRoutes(mpr)
+	tsURL, err := url.Parse(ts.URL)
+	assert.NoError(t, err)
+
+	recipient := &RecipientInfo{
+		URL:   tsURL.Host,
+		Token: "idontneedoneImtesting",
+	}
+	recipients := []*RecipientInfo{recipient}
+
+	sendFileOpts := &SendOptions{
+		DocID:   fileDoc.ID(),
+		DocType: fileDoc.DocType(),
+		Type:    consts.FileType,
+		Path: fmt.Sprintf("/sharings/doc/%s/%s", fileDoc.DocType(),
+			fileDoc.ID()),
+		Recipients: recipients,
+	}
+
+	err = SendFile(testInstance, sendFileOpts, fileDoc)
+	assert.NoError(t, err)
+}
+
+func TestSendDir(t *testing.T) {
+	fs := testInstance.VFS()
+	dirDoc := createDir(t, fs, "testsenddir")
+
+	mpr := map[string]func(*echo.Group){
+		"/sharings": func(router *echo.Group) {
+			router.POST("/doc/:doctype/:docid", func(c echo.Context) error {
+				assert.Equal(t, dirDoc.DocType(), c.Param("doctype"))
+				assert.Equal(t, dirDoc.ID(), c.Param("docid"))
+				assert.Equal(t, consts.DirType, c.QueryParam("Type"))
+				assert.Equal(t, dirDoc.DocName, c.QueryParam("Name"))
+				dirTags := strings.Join(dirDoc.Tags, files.TagSeparator)
+				assert.Equal(t, dirTags, c.QueryParam("Tags"))
+				dirDocPath, err := dirDoc.Path(fs)
+				assert.NoError(t, err)
+				assert.Equal(t, dirDocPath, c.QueryParam("Path"))
+				return c.JSON(http.StatusOK, nil)
+			})
+		},
+	}
+	if ts != nil {
+		ts.Close()
+	}
+	ts = setup.GetTestServerMultipleRoutes(mpr)
+	tsURL, err := url.Parse(ts.URL)
+	assert.NoError(t, err)
+
+	recipient := &RecipientInfo{
+		URL:   tsURL.Host,
+		Token: "idontneedoneImtesting",
+	}
+	recipients := []*RecipientInfo{recipient}
+
+	sendDirOpts := &SendOptions{
+		DocID:   dirDoc.ID(),
+		DocType: dirDoc.DocType(),
+		Type:    consts.FileType,
+		Path: fmt.Sprintf("/sharings/doc/%s/%s", dirDoc.DocType(),
+			dirDoc.ID()),
+		Recipients: recipients,
+	}
+
+	err = SendDir(testInstance, sendDirOpts, dirDoc)
+	assert.NoError(t, err)
+}
+
+func TestUpdateOrPatchFile(t *testing.T) {
+	fs := testInstance.VFS()
+	fileDoc := createFile(t, fs, "original", "original file")
+	updatedFileDoc := createFile(t, fs, "update", "this is an update")
+
+	mpr := map[string]func(*echo.Group){
+		"/files": func(router *echo.Group) {
+			router.GET("/:file-id", func(c echo.Context) error {
+				assert.NotEmpty(t, c.Param("file-id"))
+				return jsonapi.Data(c, http.StatusOK, &file{fileDoc}, nil)
+			})
+		},
+		"/sharings": func(router *echo.Group) {
+			router.PATCH("/doc/:doctype/:docid", func(c echo.Context) error {
+				assert.Equal(t, fileDoc.ID(), c.Param("docid"))
+				assert.Equal(t, fileDoc.Rev(), c.QueryParam("rev"))
+				assert.Equal(t, consts.FileType, c.QueryParam("Type"))
+
+				var patch vfs.DocPatch
+				_, err := jsonapi.Bind(c.Request(), &patch)
+				assert.NoError(t, err)
+				assert.Equal(t, fileDoc.DocName, *patch.Name)
+				assert.Equal(t, fileDoc.DirID, *patch.DirID)
+				assert.Equal(t, fileDoc.Tags, *patch.Tags)
+				assert.Equal(t, fileDoc.UpdatedAt, *patch.UpdatedAt)
```

Comment:
Could you try to compare the `Unix()` timestamp value of these two dates to fix the test ?